libs/wingsdk/src/cloud/api.ts

```diff
@@ -219,6 +209,10 @@ export abstract class Api extends Resource {
   protected corsOptions?: ApiCorsOptions;
 
   constructor(scope: Construct, id: string, props: ApiProps = {}) {
+    if (new.target === Api) {
+      return App.of(scope).newAbstract(API_FQN, scope, id, props);
+    }
+
     super(scope, id);
```

Comment:
@eladb @Chriscbr
This seems to work... Am I missing something? I feel like I remember some issues with this approach back in the polycon days

Reply:
I tried racking my brain but I couldn't think of anything concrete... curious if @eladb thinks of anything.

I'm wondering if someone tries instantiating this construct in TypeScript outside of a Wing `App` (e.g. inside of a plain cdktf `App`), maybe we can give a nicer error (e.g. "This is an abstract class, so it can only be instantiated in the scope of a Wing App")